### PR TITLE
[MOBILE-1330] add screen tracking to unity

### PR DESCRIPTION
**Pull request by crow**
### What do these changes do?
Add screen tracking to unity.

### Why are these changes necessary?
To add screen tracking to unity and reach functional parity with other plugins.

### How did you verify these changes?
Ran the sample with the unity sample script with onStart screen tracking added. Ensured breakpoint at trackScreen was hit on start when main camera was tracked.

#### Verification Screenshots:
Screen Shot 2020-02-28 at 4 24 06 PM

### CODE REVIEW

File: Assets/UrbanAirship/Platforms/iOS/UAirshipPluginIOS.cs [hunk 1]
```diff
@@ -51,9 +51,13 @@ class UAirshipPluginIOS : IUAirshipPlugin {
         [DllImport ("__Internal")]
         private static extern void UAUnityPlugin_setBackgroundLocationAllowed (bool allowed);
 
+        //Analytics Function Imports
```
Comment (BrianBatchelder) [inline Assets/UrbanAirship/Platforms/iOS/UAirshipPluginIOS.cs]:
```suggestion
        // Analytics Function Imports
```

File: Assets/Plugins/iOS/UAUnityPlugin.m [hunk 2]
```diff
@@ -241,6 +245,13 @@ void UAUnityPlugin_addCustomEvent(const char *customEvent) {
     [[UAirship shared].analytics addEvent:ce];
 }
 
+void UAUnityPlugin_trackScreen(const char *screenName) {
+    NSString *screenNameString = [NSString stringWithUTF8String:screenName];
+    UA_LDEBUG(@"UnityPlugin trackScreen");
```
Comment (BrianBatchelder) [inline Assets/Plugins/iOS/UAUnityPlugin.m]:
```suggestion
    UA_LDEBUG(@"UnityPlugin trackScreen: %@", screenName);
```

Comment (marc-scig) [reply Assets/Plugins/iOS/UAUnityPlugin.m]: I am fairly certain the `%@` format will not work for const char, so if you end up adding this, you'll probably want `%s`.

Comment (marc-scig) [reply Assets/Plugins/iOS/UAUnityPlugin.m]: or just use screenNameString, whatever's clever

Comment (BrianBatchelder) [reply Assets/Plugins/iOS/UAUnityPlugin.m]: Agreed.

File: unity-plugin/src/main/java/com/urbanairship/unityplugin/UnityPlugin.java [hunk 3]
```diff
@@ -221,6 +221,17 @@ public void addCustomEvent(String eventPayload) {
         UAirship.shared().getAnalytics().addEvent(eventBuilder.build());
     }
 
+    public void trackScreen(String screenName) {
+        PluginLogger.debug("UnityPlugin trackScreen: " + screenName);
```
Comment (marc-scig) [inline unity-plugin/src/main/java/com/urbanairship/unityplugin/UnityPlugin.java]: if you do this underneath the conditional below, we'll avoid a crash if screenName happens to be null

File: Assets/Plugins/iOS/UAUnityPlugin.m (hunk 2 above)
Comment (crow) [reply Assets/Plugins/iOS/UAUnityPlugin.m]: fixed with screenNameString

File: unity-plugin/src/main/java/com/urbanairship/unityplugin/UnityPlugin.java (hunk 3 above)
Comment (crow) [reply unity-plugin/src/main/java/com/urbanairship/unityplugin/UnityPlugin.java]: good catch

**Pull request merged**